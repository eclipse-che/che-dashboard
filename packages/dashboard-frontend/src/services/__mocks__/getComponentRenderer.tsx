/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

export default function <T extends Array<unknown>>(
  getComponent: (..._args: T) => React.ReactElement,
): {
  createSnapshot: (...args: T) => renderer.ReactTestRenderer;
  renderComponent: (...args: T) => {
    reRenderComponent: (..._args: T) => void;
  };
} {
  return {
    createSnapshot: (...args: T) => {
      return renderer.create(getComponent(...args));
    },
    renderComponent: (...args: T) => {
      const res = render(getComponent(...args));
      return {
        reRenderComponent: (..._args: T) => {
          res.rerender(getComponent(..._args));
        },
      };
    },
  };
}
