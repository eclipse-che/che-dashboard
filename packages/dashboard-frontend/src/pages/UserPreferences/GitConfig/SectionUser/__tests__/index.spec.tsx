/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import * as React from 'react';

import getComponentRenderer, { screen } from '@/services/__mocks__/getComponentRenderer';

import { GitConfigSectionUser, Props } from '..';

jest.mock('../Email');
jest.mock('../Name');

const mockOnChange = jest.fn();

const { createSnapshot, renderComponent } = getComponentRenderer(getComponent);

describe('GitConfigSectionUser', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('snapshot', () => {
    const snapshot = createSnapshot({
      config: {
        name: 'user',
        email: 'user@che',
      },
      onChange: mockOnChange,
    });
    expect(snapshot.toJSON()).toMatchSnapshot();
  });

  it('should handle name change', () => {
    renderComponent({
      config: {
        name: 'user',
        email: 'user@che',
      },
      onChange: mockOnChange,
    });

    screen.getByText('Change Name').click();

    expect(mockOnChange).toHaveBeenCalledWith({
      name: 'new user',
      email: 'user@che',
    });
  });

  it('should handle email change', () => {
    renderComponent({
      config: {
        name: 'user',
        email: 'user@che',
      },
      onChange: mockOnChange,
    });

    screen.getByText('Change Email').click();

    expect(mockOnChange).toHaveBeenCalledWith({
      name: 'user',
      email: 'new-user@che',
    });
  });
});

function getComponent(props: Props): React.ReactElement {
  return <GitConfigSectionUser {...props} />;
}
