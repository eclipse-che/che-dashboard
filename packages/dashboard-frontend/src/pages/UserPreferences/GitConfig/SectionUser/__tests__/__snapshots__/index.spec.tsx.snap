// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitConfigSectionUser snapshot 1`] = `
<div
  className="pf-c-panel"
>
  <div
    className="pf-c-panel__main"
    style={
      {
        "--pf-c-panel__main--MaxHeight": undefined,
      }
    }
  >
    <div
      className="pf-c-panel__main-body"
    >
      <form
        className="pf-c-form pf-m-horizontal"
        noValidate={true}
        onSubmit={[Function]}
      >
        <section
          aria-labelledby="pf-form-section-title0"
          className="pf-c-form__section"
          label="user"
          role="group"
        >
          <div
            className="pf-c-form__section-title"
            id="pf-form-section-title0"
          >
            [user]
          </div>
          <button
            onClick={[Function]}
          >
            Change Name
          </button>
          <button
            onClick={[Function]}
          >
            Change Email
          </button>
        </section>
      </form>
    </div>
  </div>
</div>
`;
